src/routes/concepts/components/event-handlers.mdx

```diff
@@ -3,68 +3,231 @@ title: Event Handlers
 
-If you need to attach an event listener to an element that is not supported by Solid's event delegation, you can use the [on:](/reference/jsx-attributes/on-and-oncapture) prefix followed by the event name, and assign it a function.
+If you need to attach an event listener to an element that is not supported by
+Solid's event delegation, such a custom event in a
+[`Custom Element`](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements),
+you can use the [`on:__`](/reference/jsx-attributes/on-and-oncapture) form.
 
 ```tsx
 <div on:customEvent={handleCustomEvent} />
 ```
 
-## Binding events
+### Event delegation peculiarities
 
-In Solid, event handlers can be bound by passing an array to the event handler.
-The second item of the array is supplied as the handler's first argument.
+While delegated events provide some performance enhancements, it comes with some
+tradeoffs that should be considered.
+
+ <Callout type="info" title="Design">
+
+Event delegation it's also designed for event propagation through the JSX tree,
+rather than the DOM tree.
+
+</Callout>
+
+This differs from the acquired expectations one may have over how events work
+and flow.
+
+Here is a list of peculiarities to keep in mind:
+
+1.  Delegated event listeners are added to the `document` once per event type.
+    When the element/component that added it is removed, it's handler function
+    will no longer be called, but the delegated event listener will still be
+    present in the `document`. Considering the following example, a `div`
+    listening for `mousemove`. `mousemove` will still be dispatched to the
+    `document` even if the `div` is disposed.
 
 ```tsx
-const handler = (data, event) => {
-	console.log("Data:", data, "Event:", event);
-};
+<div onMouseMove={handleCustomEvent} />
+```
 
-<button onClick={[handler, "Hello!"]}>Click Me</button>;
+<Callout type="tip" title="Ocassional Events">
+Consider using Native Events instead of Delegated Events for rare or ocassional events that happen on special circumstances and won't benefit from the performance improvements of event delegation.
+
+```tsx
+<div on:mousemove={handleCustomEvent} />
+```
+
+  </Callout>
+
+2. When in need to `stopPropagation` use a Native Event instead.
+   `event.stopPropagation()` won't work as expected as the event is attached to
+   the `document` instead of to the `element`.
+
+   Considering the following example. Using a Native Event would have stopped
+   the event from reaching the `div native` handler. This is not the case for
+   Delegated Events.
+
+```tsx
+onMount(() => {
+  ref.addEventListener("click", () => {
+    console.log("div native");
+  });
+});
+<div ref={ref}>
+  <button
+    onClick={(event) => {
+      event.stopPropagation();
+      console.log("button");
+    }}
+  >
+    button
+  </button>
+</div>;
+```
+
+This can be solved by switching the `button` event to use a Native Event as
+follows:
+
+```tsx
```

Comment:
This will mark/highlight the change, just makes it a little more obvious
```suggestion
```tsx ins="on:click"
```